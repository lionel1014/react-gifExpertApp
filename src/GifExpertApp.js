import React , {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = () => {

    //const categories = ['DMC','Resident Evil','Wow'];
    const [categories, setCategories] = useState(['Gamer']);

    // const handleAdd = () =>{
    //     setCategories([...categories,'Lol']);
    //     // setCategories( cate => [...cate,'Lol']);
    // };
    return (
        <>
        <h2>GifExpertApp (buscador de gif's) </h2>
        <AddCategory setCategories= {setCategories} />
        <hr />

        <ol>
            {//las llaves marcan una expresion en javascrip
                categories.map ( (category) =>( 
                    <GifGrid  
                    key = {category}
                    category = {category} 
                    /> 
                ))  
                // lo que encuentra lo guarda en category //la funcion map busca todas las categorias y su indice
                
            }
        </ol>

        </>
    );

}



//rafc