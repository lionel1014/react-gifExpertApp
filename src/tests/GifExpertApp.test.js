import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Pruebas en nuestro componente PADRE <GifExpert App />', () => {
    
    test('Debe mostrar correctamente el componente (Snapshot) ', () => {
        
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe mostrar una lista de Categorias ', () => {
        
        const categories = ['Pepe','Gamer'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories} /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length); //preguntamos si hay las mismas categorias que se invento

    })
    
});
