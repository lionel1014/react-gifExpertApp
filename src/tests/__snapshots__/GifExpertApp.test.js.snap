// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en nuestro componente PADRE <GifExpert App /> Debe mostrar correctamente el componente (Snapshot)  1`] = `
<Fragment>
  <h2>
    GifExpertApp (buscador de gif's) 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en nuestro componente PADRE <GifExpert App /> Debe mostrar una lista de Categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp (buscador de gif's) 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Pepe"
      key="Pepe"
    />
    <GifGrid
      category="Gamer"
      key="Gamer"
    />
  </ol>
</Fragment>
`;
